internal/dataapi/server/insert_one.go

```diff
@@ -47,11 +48,19 @@ func (s *Server) InsertOne(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var documents any
-	if insertDoc != nil {
-		documents = wirebson.MustArray(insertDoc)
+	if _, ok := insertDoc.(wirebson.AnyDocument); !ok {
```

Comment:
You could set `doc, ok := insertDoc.(wirebson.AnyDocument)`, because it's used by `ensureId` parameter.